samples/app-anonymous-users/csharp/ClientApp/src/components/shareview.jsx

```diff
@@ -55,40 +73,239 @@ const ShareView = () => {
+    const verifyAnonymousUser = () => {
+          microsoftTeams.app.getContext().then((context) => {
+            if (context.user.licenseType === "Anonymous") {
+                setIsAnonymousUser(true);
+            }
+        });
+    }
+
     // Click submit vote button
     const submitVote = () => {
-        if (connection) {
-            microsoftTeams.app.getContext().then((context) => {
 
-                setDisabled(true); // Disable Button
-                // Once we call getContext API, we can recognize anonymous users by checking for the licenseType value like: context.user.licenseType === "Anonymous".
-                if (context.user.licenseType === "Anonymous")
-                {
-                    // Update the state property for incremented count value.
-                    let addAnonymousVal = aShowCount + 1;
+        if (connection) {
+            setIsVoteBtnDisabled(true); // Disable Button
+            if (IsAnonymousUser) {
+                // Update the state property for incremented count value.
+                let addAnonymousVal = aShowCount + 1;
 
-                    // Sending the updated count to hub signal to show the latest data on stage view.
-                    connection.send("SendMessage", "Anonymous", addAnonymousVal);
-                }
-                else {
-                    // Update the state property for incremented count value.
-                    let addUserVal = uShowCount + 1;
+                // Sending the updated count to hub signal to show the latest data on stage view.
+                connection.send("SendMessage", "Anonymous", addAnonymousVal);
+            }
+            else {
+                // Update the state property for incremented count value.
+                let addUserVal = uShowCount + 1;
 
-                    // Sending the updated count to hub signal to show the latest data on stage view.
-                    connection.send("SendMessage", "User", addUserVal);
-                }
-            });
+                // Sending the updated count to hub signal to show the latest data on stage view.
+                connection.send("SendMessage", "User", addUserVal);
+            }
         }
         else {
             alert('No connection to server yet.');
         }
     }
 
+    // Initiate facebook login.
+    const facebookLogin = () => {
+        fbAuthentication() // This method get a client-side token for Facebook
+            .then((result) => {
+                return getFacebookProfileName(result.idToken); // This method get a face book user profile details.
+            })
+            .catch((error) => {
+                console.log(error);
+            });
+    }
+
+    // Get client side token for facebook.
+    const fbAuthentication = () => {
+        var facebookAppId = process.env.REACT_APP_FACEBOOK_APP_ID;
+        let redirectUri = window.location.origin + "/facebook-auth-end";
+        let state = Math.random().toString(36).substring(2, 7);
```

Comment:
Have you updated this same to use same card flow as done in NodeJS sample for sending notifications?

Reply:
Yes i have done same

Reply:
Will raise another small PR for that.